### PR TITLE
Unplugging MaterialHelper

### PR DESCRIPTION
**Pull request by deltakosh**
Overall idea is to move big contenders out of ParticleSystem and create a ThinParticleSystem with the minimum rendering loop in it

This is first step with MaterialHelper

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14804/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14804/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14804/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14804/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14804/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14804/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14804/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14804/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14804/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14804/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/scene.constants.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Class used to store scene constants
+ */
+export class SceneConstants {
```
Comment (RaananW) [inline packages/dev/core/src/scene.constants.ts:4]: if we are already reducing clutter, maybe make that a const object and not a class? no need for a class with static members only.

**Review by RaananW — APPROVED** [review]
You know my opinion about this ;-)

File: packages/dev/core/src/scene.constants.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/scene.constants.ts:4]: sounds good!

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14804/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by RaananW** [comment]
External packages that need the functions can't find it. Probably a missing export?

**Review by Popov72 — APPROVED** [review]

**Pull request merged**